src/qtt/algorithms/markov_chain.py

```diff
@@ -59,10 +62,10 @@ def generate_choice(self) -> int:
             Integer in the range 0 to the number of states
         """
         self._idx = self._idx + 1
-        if self._idx == self._block.size:
+        if self._idx == self.block_size:
             self._idx = 0
-            self._block = self._generate_block()
-        return int(self._block[self._idx])
+            self._block = self._generate_block().tolist()
```

Comment:
To see this working (twice) the unit test `test_ChoiceGenerator` in test_markov_chain.py, line 17 should be range(10000) (bigger than default block_size 5000)